### ISSUE TITLE
no-reserved-keywords not finding "catch" in all places

### ISSUE THREAD

**Issue by chrismbarr** [microsoft/tslint-microsoft-contrib#395]
I have this code:
```ts
spyOn(customerService, "getCustomer").and.returnValue({
  then: (): {catch: () => void} => {
    return {
      catch: (): void => { return; }
    };
  }
});
```

Running with
- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__:  gulp

#### Actual behavior

It only reports a violation in the typedef

![image](https://user-images.githubusercontent.com/463685/31462378-94a0e968-ae9a-11e7-85f1-44ad8f7e23b7.png)

#### Expected behavior
I would expect it to also report a violation in the return statement below where the actual object property is named `catch`, in addition to the typedef above.

**Comment by chrismbarr** [comment]
Ok, what you've said makes sense.  However, if the goal is not to use a reserved JS keyword in TS code I'd say that only flagging violations in typedefs is somewhat useless since that's not real "code" - meaning it never makes it to the transpiled output.

To me, it makes sense to flag the reserved keyword in both instances so that it's not used in real code, and also in the typedefs since that might help you catch this early as you typically write typedefs before you write the actual code and you get a change to name a better naming choice earlier.

If I write:
```ts
interface IMyThing {
  var: string;
  type: string;
  catch: (err: string) => void;
}

const someObj: IMyThing = {
  var: 'myNum',
  type: 'number',
  catch: (err: string) => console.log(err)
};
```

I'd expect to see violations on the interface and the usage since both use the same property name we've deemed to be non-ideal and possibly misleading.

For systems already in place, renaming properties is probably difficult (like in your attached PR).  However, in places were it's easier to change, having one violation per usage might help give you an idea of how widely spread this mis-named property is, and that would be encouragement for renaming in.

Here's how I'd re-write my above example
```ts
interface IMyThing {
  variableName: string;
  variableType: string;
  onError: (err: string) => void;
}

const someObj: IMyThing = {
  variableName: 'myNum',
  variableType: 'number',
  onError: (err: string) => console.log(err)
};
```

**Comment by JoshuaKGoldberg** [comment]
How about an opt-in config option to add the extra checks? Agreed both that you should be allowed to use these _if_ you need, but that it's better to avoid them.

**Comment by chrismbarr** [comment]
I think that way makes the most sense.  The default doesn't work for everyone, so let it be configured!

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
